File: 0896-monotonic-array/0896-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        
                if(nums.length<2) return true;

        int i=0;
        while(nums[i]==nums[i+1]){
            if ( i==nums.length-2) return true;
            i++;
        } 
        
        
        
        if(nums[i]<nums[i+1]){
            for (int j=1;j<nums.length-1;j++){
                if(nums[j]>nums[j+1]) return false;
            }return true;
        }else if(nums[i]>nums[i+1]){
            for(int j=1;j<nums.length-1;j++){
                if (nums[j]<nums[j+1]) return false;
            }return true;
        }
        return true;
    }
}

File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        bool inc=false;
        bool dec=false;
        for (int i=0;i<nums.size()-1;i++){
            if (nums[i]<nums[i+1]) inc=true;
            if (nums[i]>nums[i+1]) dec=true;
        }
        if (inc==true&& dec==true) return false;
        return true;
    }
};